### PR TITLE
mozilla-mobile/fenix#10261: pwa telemetry

### PR DESCRIPTION
**Pull request by eliserichards**
---

Last PR for this, I swear 😅 

Whew. Ok. Foreground and background events are in Fenix, in their rightful homes (_note to self: don't put logic into the sample browser_). The `WebAppLauncherActivity` is handling homescreen taps and looking for a manifest to determine if it's a PWA. Install facts are ok where they are now, in the `WebAppShortcutManager`.

### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

**Comment by eliserichards** [comment]
bors try

File: samples/browser/src/main/java/org/mozilla/samples/browser/ExternalAppBrowserFragment.kt [hunk 1]
```diff
@@ -101,6 +107,18 @@ class ExternalAppBrowserFragment : BaseBrowserFragment(), UserInteractionHandler
         return layout
     }
 
+    override fun onResume() {
+        super.onResume()
+        val currTime = SystemClock.elapsedRealtimeNanos()
+        emitForegroundTimingFact(currTime)
```
Comment (NotWoods) [inline samples/browser/src/main/java/org/mozilla/samples/browser/ExternalAppBrowserFragment.kt:113]: Quick breakdown of our browsers:
- Sample Browser (this one): Demo of our Android Components
- Reference Browser: Starting point for building a full browser with Android Components
- Fenix: some product idk

All 3 have their own browser fragments and external app browser fragments. If you want to add telemetry for Fenix it should go in the Fenix codebase :)

Comment (eliserichards) [reply samples/browser/src/main/java/org/mozilla/samples/browser/ExternalAppBrowserFragment.kt:113]: 😰 dang this is confusing hahahaha

File: components/feature/pwa/src/main/java/mozilla/components/feature/pwa/WebAppLauncherActivity.kt [hunk 2]
```diff
@@ -67,7 +54,10 @@ class WebAppLauncherActivity : AppCompatActivity() {
         when (manifest?.display) {
             WebAppManifest.DisplayMode.FULLSCREEN,
             WebAppManifest.DisplayMode.STANDALONE,
-            WebAppManifest.DisplayMode.MINIMAL_UI -> launchWebAppShell(startUrl)
+            WebAppManifest.DisplayMode.MINIMAL_UI -> {
+                emitHomescreenIconTapFact()
```
Comment (NotWoods) [inline components/feature/pwa/src/main/java/mozilla/components/feature/pwa/WebAppLauncherActivity.kt:58]: You can leave `emitHomescreenIconTapFact` in `launchWebAppShell` where it was instead. I'm fine with either location.

File: components/lib/crash/docs/metrics.md [hunk 3]
```diff
@@ -18,10 +18,12 @@ See the Glean SDK documentation for the [`metrics` ping](https://mozilla.github.
 
 The following metrics are added to the ping:
 
-| Name | Type | Description | Data reviews | Extras | Expiration |
-| --- | --- | --- | --- | --- | --- |
-| crash_metrics.crash_count |[labeled_counter](https://mozilla.github.io/glean/book/user/metrics/labeled_counters.html) |Counts the number of crashes that occur in the application. This measures only the counts of each crash in association with the labeled type of the crash. The labels correspond to the types of crashes handled by lib-crash. Deprecated: `native_code_crash` replaced by `fatal_native_code_crash` and `nonfatal_native_code_crash`  |[1](https://bugzilla.mozilla.org/show_bug.cgi?id=1553935#c3), [2](https://github.com/mozilla-mobile/android-components/pull/5700#pullrequestreview-347721248)|<ul><li>uncaught_exception</li><li>caught_exception</li><li>fatal_native_code_crash</li><li>nonfatal_native_code_crash</li></ul>|never |
+| Name | Type | Description | Data reviews | Extras | Expiration | [Data Sensitivity](https://wiki.mozilla.org/Firefox/Data_Collection) |
```
Comment (NotWoods) [inline components/lib/crash/docs/metrics.md:21]: Let's remove the metrics docs from the PR, gl has a separate PR to commit these

Comment (eliserichards) [reply components/lib/crash/docs/metrics.md:21]: These were auto generated when I built. I can try to remove them 🤔

**Comment by eliserichards** [comment]
bors r=NotWoods

**Pull request merged**